packages/notifications/src/useNotifications.ts

```diff
@@ -0,0 +1,36 @@
+import { createContext, createElement, PropsWithChildren, useContext } from 'react';
+
+export type NotificationType = 'error' | 'warn' | 'info' | 'success' | 'welcome';
+export interface NotificationOptions {
+  detail: string | unknown;
+}
+
+export interface NotificationsContract {
+  notify: (type: NotificationType, title: string, options?: NotificationOptions) => void;
+}
+const NotificationsContext = createContext<NotificationsContract | null>(null);
+
+export type NotificationsProviderProps = PropsWithChildren<{
+  notifications: NotificationsContract;
+}>;
+
+export const text = {
+  error: {
+    noProvider:
+      'No NotificationsProvider provided. Components (or hooks within them) using useNotificationsFromContext must be descendants of NotificationsProvider.',
+  },
+};
+
+/** Provides notifications to descendents via React Context. */
+export const NotificationsProvider = (props: NotificationsProviderProps) => {
```

Comment:
Super nit: convention.

```suggestion
export const NotificationsProvider = (props: NotificationsProviderProps): ReactNode => {
```

Reply:
ah, this was a copy-and-modify of ThemeProvider, just checked and it lacks the ReactNode return convention also.  I can add to both.